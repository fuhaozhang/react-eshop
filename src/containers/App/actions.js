import { LOAD_APP, ADD_TO_CART, TRANSFER_TO_USER_CART, LOAD_CART, SET_CART } from './constants';

export function addToCart(product) {
  return {
    type: ADD_TO_CART,
    payload: { product },
  };
}

export function transferToUserCart() {
  return {
    type: TRANSFER_TO_USER_CART,
  };
}

export function loadCart() {
  return {
    type: LOAD_CART,
  };
}

export function setCart(cart) {
  return {
    type: SET_CART,
    payload: { cart },
  };
}
