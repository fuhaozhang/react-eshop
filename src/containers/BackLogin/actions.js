import { LOGIN, LOGIN_SUCCESS, LOGIN_FAIL, SET_CURRENT_USER } from './constants';

export function login(email, password) {
  return {
    type: LOGIN,
    payload: {
      email,
      password
    }
  };
}

export function loginSuccess(user) {
  return {
    type: LOGIN_SUCCESS,
    payload: user
  };
}

export function loginFail(err) {
  return {
    type: LOGIN_FAIL,
    payload: err
  };
}

export function setCurrentUser(user) {
  return {
    type: SET_CURRENT_USER,
    payload: user
  };
}