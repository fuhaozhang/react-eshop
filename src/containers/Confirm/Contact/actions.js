import { FETCH_CONTACTS, SET_CONTACTS, SAVE_CONTACT, SET_ERROR } from './constants';

export function fetchContacts() {
  return {
    type: FETCH_CONTACTS,
  };
}

export function setContacts(contacts) {
  return {
    type: SET_CONTACTS,
    payload: {
      contacts,
    },
  };
}

export function setError(error) {
  return {
    type: SET_ERROR,
    payload: {
      error,
    },
  };
}

export function saveContact(contact) {
  return {
    type: SAVE_CONTACT,
    payload: {
      contact,
    },
  };
}
