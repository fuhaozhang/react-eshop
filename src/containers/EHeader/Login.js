import React, { useState, useRef } from 'react';
import classnames from 'classnames';
import _ from 'lodash';
import { Modal } from 'antd';

import EInput from '@/components/EInput';
import $style from './Login.module.scss';

function Login({ visible, onCancel }) {
  let loginRef = useRef();

  let [email, setEmail] = useState('');

  return (
    <div className={$style.login} ref={loginRef}>
      <Modal visible={visible} footer={null} width="386px" getContainer={() => loginRef.current} onCancel={onCancel}>
        <div className={$style.login__content}>
          <div className={$style.login__header}>
            <div className={$style.login__type}>手机号登陆</div>
            <div className={$style.login__split}>&nbsp;</div>
            <div className={classnames($style.login__type, $style.login__type_active)}>邮箱登陆</div>
          </div>
          <div className={$style.login__bodyEmail}>
            <form>
              <EInput className={$style.login__email} prefix={<img src={require('@/assets/login/user.png')} />} placeholder="邮箱账号" />
              <EInput className={$style.login__password} prefix={<img src={require('@/assets/login/lock.png')} />} placeholder="密码" />
              <button className={$style.login__submit}>登陆</button>
            </form>
          </div>
          <div className={$style.login__footer}>
            <div className={$style.social}>
              <a className={classnames($style.social__icon, $style.social__weixin)}></a>
              <a className={classnames($style.social__icon, $style.social__qq)}></a>
              <a className={classnames($style.social__icon, $style.social__weibo)}></a>
              <a className={classnames($style.social__icon, $style.social__wangyi)}></a>
            </div>
            <div className={$style.help}>
              <span className={$style.register}>邮箱注册</span>
              <span className={$style.forget}>忘记密码</span>
            </div>
          </div>
        </div>
      </Modal>
    </div>
  );
}

export default Login;
