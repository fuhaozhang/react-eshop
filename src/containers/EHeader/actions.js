import { LOAD } from './constants';

export function load() {
  return {
    type: LOAD,
  };
}
