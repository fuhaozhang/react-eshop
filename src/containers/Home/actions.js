import { FETCH_CAROUSELS, SET_CAROUSELS } from './constants';

export function fetchCarousels() {
  return {
    type: FETCH_CAROUSELS,
  };
}

export function setCarousels(carousels) {
  return {
    type: SET_CAROUSELS,
    payload: {
      carousels,
    },
  };
}
