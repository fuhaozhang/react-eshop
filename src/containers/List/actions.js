import { FETCH_CAROUSELS, SET_CAROUSELS, FETCH_DATA, SET_CATEGORY, SET_THIRD_CATEGORIES, SET_DATA } from './constants';

export function fetchCarousels(categoryId) {
  return {
    type: FETCH_CAROUSELS,
    payload: {
      categoryId,
    },
  };
}

export function setCarousels(carousels) {
  return {
    type: SET_CAROUSELS,
    payload: {
      carousels,
    },
  };
}

export function fetchData(categoryId) {
  return {
    type: FETCH_DATA,
    payload: {
      categoryId,
    },
  };
}

export function setCategory(category) {
  return {
    type: SET_CATEGORY,
    payload: {
      category,
    },
  };
}

export function setThirdCategories(thirdCategories) {
  return {
    type: SET_THIRD_CATEGORIES,
    payload: {
      thirdCategories,
    },
  };
}

export function setData(data) {
  return {
    type: SET_DATA,
    payload: {
      data,
    },
  };
}
