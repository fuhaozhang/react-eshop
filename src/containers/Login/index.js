import React, { useState } from 'react';
import { Alert, Form, Input, Divider, Row, Button, Icon, Col } from 'antd';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import _ from 'lodash';
import classnames from 'classnames';

import { makeSelectError, makeSelectIsLogining } from './selectors';
import { login } from './actions';

import Header from './Header';
import Footer from './Footer';
import LoginMethod from './LoginMethod';
import $style from './index.module.scss';

import qqIcon from '@/assets/login/qq.png';
import weixinIcon from '@/assets/login/weixin.png';
import weiboIcon from '@/assets/login/weibo.png';
import zhifubaoIcon from '@/assets/login/zhifubao.png';

class Login extends React.Component {
  render() {
    const { 
      error,
      submitForm,
      isLogining,
      form 
    } = this.props;

    const { getFieldDecorator, setFieldsValue, validateFields } = form;

    let errorMessage = null;
    if (error) {
      errorMessage = <Alert className={classnames('mb4')} message={error} type="error" closable />
    }

    const onSubmitForm = (evt) => {
      if (isLogining) return;

      evt && evt.preventDefault();

      validateFields((err, values) => {
        if (!err) {
          submitForm(values.email, values.password);
        }
      });
    };

    return (
      <div >
        <Header></Header>
        <div className={$style.banner}>
          <div className={classnames('container', $style.banner__content)}>
            <Row>
              <Col lg={16} md={12}><img className={$style.banner__img} src={require('@/assets/login/banner.png')} /></Col>
              <Col lg={8} md={12}>
                <div className={$style.panel}>
                  <div className={$style.panel__header}>
                    <img className={$style.logo} src={require('@/assets/logo.png')} />
                    <div className={$style.title}>欢迎登陆eshop</div>
                  </div>
                  {errorMessage}
                  <div className={$style.panel__body}>
                    <Form onSubmit={onSubmitForm}>
                      <Form.Item>
                        {getFieldDecorator('email', {
                          rules: [
                            {
                              required: true,
                              message: '请输入邮箱!',
                            },
                          ],
                        })(<Input
                          prefix={<Icon type="mail" />}
                          placeholder="邮箱"
                          allowClear
                          onChange={(evt) => setFieldsValue({ email: evt.target.value })}
                        />)}
                      </Form.Item>

                      <Form.Item>
                        {getFieldDecorator('password', {
                          rules: [
                            {
                              required: true,
                              message: '请输入密码!',
                            },
                          ],
                        })(<Input.Password
                          prefix={<Icon type="lock" />}
                          visibilityToggle
                          placeholder="密码"
                          allowClear
                          onChange={(evt) => setFieldsValue({ password: evt.target.value })}
                        />)}
                      </Form.Item>

                      <Row className={classnames('mb4')} type="flex" justify="end">
                        <Button className={$style.header__feedback} type="link">忘记密码?</Button>
                        <Button className={$style.header__feedback} type="link">注册</Button>
                      </Row>
                      <Button htmlType="submit" type="danger" loading={isLogining} block>
                        登陆
                    </Button>
                    </Form>
                    <Divider></Divider>
                    <div className={$style.others}>
                      <span className={$style.others__title}>其他登录方式:</span>
                      <LoginMethod className={$style.others__icon} title="QQ">
                        <img src={qqIcon} />
                      </LoginMethod>
                      <LoginMethod className={$style.others__icon} title="微信">
                        <img src={weixinIcon} />
                      </LoginMethod>
                      <LoginMethod className={$style.others__icon} title="微博">
                        <img src={weiboIcon} />
                      </LoginMethod>
                      <LoginMethod className={$style.others__icon} title="支付宝">
                        <img src={zhifubaoIcon} />
                      </LoginMethod>
                      <LoginMethod className={$style.others__icon} title="手机号登录">
                      </LoginMethod>
                    </div>
                  </div>
                </div></Col>
            </Row>
          </div>
        </div>
        <Footer></Footer>
      </div>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  error: makeSelectError(),
  isLogining: makeSelectIsLogining(),
});

export function mapDispatchToProps(dispatch) {
  return {
    submitForm: (email, password) => dispatch(login(email, password))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Form.create({ name: 'login' })(Login));

