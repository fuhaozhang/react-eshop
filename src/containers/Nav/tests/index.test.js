import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import configureStore from '@/store';

import { mapDispatchToProps } from '../index';
import { Nav } from '../index';
import { showLogin, logout } from '@/containers/Login/actions';
import { fetchNotices } from '../actions';

describe('<Nav />', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  it('snapshot', () => {
    const loadSpy = jest.fn();
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <Nav user={null} notices={null} onLoad={loadSpy} />
      </Provider>
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('调用加载函数', () => {
    const loadSpy = jest.fn();
    render(
      <Provider store={store}>
        <Nav onLoad={loadSpy} />
      </Provider>
    );
    expect(loadSpy).toHaveBeenCalled();
  });

  describe('dispatch', () => {
    it('调用onShowLogin时dispath showLogin', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      result.onShowLogin();
      expect(dispatch).toHaveBeenCalledWith(showLogin());
    });

    it('调用onLoad时dispath showLogin', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      result.onLoad();
      expect(dispatch).toHaveBeenCalledWith(fetchNotices());
    });

    it('调用onLogout时dispath logout', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      result.onLogout();
      expect(dispatch).toHaveBeenCalledWith(logout());
    });
  });
});
