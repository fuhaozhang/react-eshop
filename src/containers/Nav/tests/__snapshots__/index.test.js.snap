// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> snapshot 1`] = `
<nav
  class="nav"
>
  <div
    class="content container"
  >
    <div
      class="notice"
    >
      <img
        class="notice__img"
        src="speaker.gif"
      />
      <ul
        class="notice__list"
      >
        <li
          class="notice__item"
        />
      </ul>
    </div>
    <div
      class="link"
    >
      <ul
        class="link__list"
      >
        <li
          class="link__item"
        >
          <a>
            登录/注册
          </a>
        </li>
        <li
          class="link__item"
        >
          <a>
            会员
          </a>
        </li>
        <li
          class="link__item"
        >
          <a>
            甄选家
          </a>
        </li>
        <li
          class="link__item"
        >
          <a
            class="ant-dropdown-trigger"
          >
            企业采购 
            <span
              aria-label="down"
              class="anticon anticon-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </a>
        </li>
        <li
          class="link__item"
        >
          <a
            class="ant-dropdown-trigger"
          >
            客户服务 
            <span
              aria-label="down"
              class="anticon anticon-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </a>
        </li>
        <li
          class="link__item"
        >
          <a>
            <span
              class="link__appIcon"
            />
            APP
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="login"
  />
</nav>
`;
